File: Data Structures & Algorithms/first-missing-positive/submission-3.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {number}
     */
    firstMissingPositive(nums) {
        const LEN = nums.length
        for(let i = 0; i < LEN; ++i) {
            while(nums[i] > 0 && nums[i] < LEN && nums[nums[i] - 1] !== nums[i]) {
                [nums[nums[i] - 1], nums[i]] = [nums[i], nums[nums[i] - 1]];
            }
        }

        for(let i = 0; i < LEN; ++i) {
            if(i + 1 !== nums[i]) return i + 1;
        }

        return LEN + 1;
    }
}

File: Data Structures & Algorithms/two-integer-sum/submission-3.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        size = len(nums)
        index_map = {}

        for i in range(size):
            diff = target - nums[i]
            if diff in index_map:
                index = index_map[diff]
                return [i, index] if i < index else [index, i]
            
            index_map[nums[i]] = i

        raise Exception("Something wrong happened")